jena-arq/src/main/java/org/apache/jena/atlas/web/TypedInputStream.java

```diff
@@ -31,6 +31,7 @@
     // May be different from the URI used to access the resource 
     // e.g. 303 redirection, mapped URI redirection 
     private String baseURI ;
+    private boolean isClosed = false ;
```

Comment:
Unnecessary in this case because it is safe to call close on the wrapped `InputStream` anyway.  The flag goes from false to true only.  The flag is useful (e.g. debugging) but is safe without. (I don't like over using `volatile` on principle.)